src/mudlet.cpp

```diff
@@ -3671,6 +3646,36 @@ void mudlet::startAutoLogin()
     }
 }
 
+bool mudlet::addHost(const QString& hostname, const QString& port, const QString& login, const QString& pass)
+{
+    auto success = mHostManager.addHost(hostname, port, login, pass);
+    if (mpDebugArea) {
+        return success;
+    }
+
+    if (success && !mpDebugArea) {
+        mpDebugArea = new QMainWindow(nullptr);
+        mpDefaultHost = mHostManager.getHost(hostname);
```

Comment:
⚠️ Um, Huston, we *may* have a problem - it looks like this and the next line will make the *Central Debug Console* belong to the *first* profile open - so what happens to the CDC when multi-playing and the first profile is *closed* - does that make the CDC die as well...?

Reply:
Eh, yes, you raise a valid point. I'll have it migrate to the next available profile (if any) then.

Reply:
Hm, tough. We could just also close the debug area and re-open it with the next available host, but then the user experience is not so nice: everything that was in the debug area will get lost. On the other hand, implementing host migration for an existing TConsole is a fair bit of work just for this one usecase.

Reply:
Does it *have* to be a `Host` pointer - or is it only for the `QObject::parent` property so that if the parent is destroyed it causes all the `QObject` derived children of that parent to commit suicide? If that is the case then perhaps the CDC can be given the `mudlet` singleton as a parent - then things ought to be fine.

I haven't got the code open in front of me right now (😮) but I will take a look over the weekend if I can make sense of this...

> On the other hand, implementing host migration for an existing TConsole is a fair bit of work just for this one usecase.

That one `TConsole` instance is an important use case - and we (I certainly) have suggested to users to have the debug console opened from a dummy profile *before* the one with a start-up problem is initiated - so that early debug output *can* be observed in the latter - so it is important that the CDC is shared equally between all open profiles. {Fair enough there is also a long-standing need for a means to show which profile produces which lines in the CDC!}

Reply:
It does need a valid `Host`, yeah.

Reply:
Fixed.